spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -168,6 +169,9 @@ protected void onInit() {
 					this.taskExecutor = new ErrorHandlingTaskExecutor(this.taskExecutor, this.errorHandler);
 				}
 			}
+			if (this.transactionSynchronizationFactory == null) {
+				this.transactionSynchronizationFactory = new PassthroughTransactionSynchronizationFactory();
+			}
```

Comment:
`TSM.iATA()` accesses a thread local - since 99.99% of pollers are __not transactional__ I'd prefer to not take two TL hits (here and in the poller's finally block) on every poll - a simple `null` check on the TSF field will be much more efficient.

Something less than 99.99%, but probably > 98% of pollers don't have an advice chain at all, so I can live with that (presence of an advice chain) being the condition we use to decide whether or not to store the message in a TL.

@andybaer You don't need to take any action until @artembilan and I have resolved our differences 😄